lib/rules/avoid-leaking-state-in-ember-objects.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+const ember = require('../utils/ember');
+const utils = require('../utils/utils');
+
+
+//------------------------------------------------------------------------------
+// Ember object rule - Avoid leaking state
+// (Don't use arrays or objects as default props)
+//------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Avoids state leakage',
+      category: 'Ember Object',
+      recommended: true
+    },
+    fixable: null, // or "code" or "whitespace"
+    schema: [{
+      type: 'array',
+      items: { type: 'string' },
+    }],
+  },
+
+  create(context) {
+    const defaultIgnoredProperties = [
+      'classNames',
+      'classNameBindings',
+      'actions',
+      'concatenatedProperties',
+      'mergedProperties',
+      'positionalParams',
+      'attributeBindings',
+      'queryParams',
+      'attrs',
+    ];
+
+    let ignoredProperties = context.options[0] || defaultIgnoredProperties;
+
+    const message = 'Only string, number, symbol, boolean, null, undefined, and function are allowed as default properties';
```

Comment:
Its probably simpler to reason about if we inline this message into the `report` function below. What do you think?